New way to close the popup player

mauriciocolli opened the pull request:
- Add an overlay at the bottom of the screen that the user can drag the popup to, closing it.

    A little demo showing how it works:
    <img src="https://user-images.githubusercontent.com/13389907/44011087-e7bbce14-9e8c-11e8-8c42-dca652fee3f7.gif" width="200"/>

- Do not show the player controllers when moving, since the user have to click in the popup
anyway now (#1342).

---
Closes #658
Related #979, #1306

karyogamy commented on app/src/main/java/org/schabi/newpipe/player/PopupVideoPlayer.java:
```diff
@@ -969,5 +1092,43 @@ private boolean handleMultiDrag(final MotionEvent event) {
+        // Utils
+        //////////////////////////////////////////////////////////////////////////*/
+
+        private float map(float value, float start1, float stop1, float start2, float stop2) {
```
Might be a good idea to add some comments here to describe what map does as well as some explanations for the constants `0x88` and `1.15f` used below as well.

mauriciocolli replied:
The`map()` was removed, and comment made about the percentage.

karyogamy commented on app/src/main/java/org/schabi/newpipe/player/PopupVideoPlayer.java:
```diff
@@ -894,21 +1014,21 @@ public boolean onFling(MotionEvent e1, MotionEvent e2, float velocityX, float ve
             final float absVelocityY = Math.abs(velocityY);
             if (absVelocityX > shutdownFlingVelocity) {
-                onClose();
+                closePopup();
```
Since now we have a proper quick close option, it'd be a good time to remove this swipe to close (shutdown fling) feature.

mauriciocolli replied:
Removed.

mauriciocolli commented:
> disable or make invisible the button when soft keyboard is open.

Is there a way to do that reliably (or at all)?

> move the button on top of the keyboard

Was the chosen solution.

But damn, was it a buggy road. I finally figured out a way to do that work: _do not_ mess with visibility of the view being added to the window manager. Maybe it was messing with the view being measured? Problem with using `MATCH_PARENT`? Well, it all sorted it out at the end...

Yet, there's still a major bug left: the popup don't have the right position if it was positioned in the keyboard area when it appears. Maybe the solution will be something similar to how the the close overlay works, a `MATCH_PARENT` view. We can use it to detect its size changing (the draggable area) and update its boundaries accordingly. Didn't mess with it yet, something for another time/PR...

> It seems excessive to have both the popup overlay turning red and the close button shaking 

I have removed the shaking. It was meant to make the closing action more obvious when the overlay was not that visible, but that doesn't happen very often (basicly just pure red videos, or whatever the color the overlay could be). I also removed the overlay alpha-distance mapping, so it even further make the shaking unnecessary.

karyogamy commented on app/src/main/java/org/schabi/newpipe/util/AnimationUtils.java:
```diff
@@ -241,6 +241,33 @@ public void onAnimationEnd(Animator animation) {
                 }).start();
     }
 
+    public static ValueAnimator shakeView(View view) {
```
This can be removed now.

karyogamy commented:
> Is there a way to do that reliably (or at all)?

There doesn't seem to be an easy way, I might be talking out of my ass when suggesting this solution =\

> But damn, was it a buggy road.

Didn't expect it to have that much resistance, what was the buggy behaviour?
But in any case, good work, I think this is good to go. 
Will be testing it out on a real device the next couple of days.

mauriciocolli commented:
> what was the buggy behaviour?

The overlay would not have the correct height, would clip through the keyboard, buggy exit animation and wouldn't have consistency across the APIs versions. Tried multiple combinations of flags and other things.

Basically, it would not have the correct dimensions, and I think the reasons are those that I mentioned earlier (views were `GONE` and wouldn't measure correctly I guess, even though I think it should when it was made `VISIBLE` again). Stupid mistake maybe...

---

> the popup don't have the right position if it was positioned in the keyboard area when it appears

And re-reading what I wrote I just had this "brilliant" idea: why not using the _overlay_ layout changes to correct the popup's position? I'll do some tests and maybe push a commit later.

karyogamy commented:
I see what you did there, good one.

theScrabi commented:
Cant you still close through the notification?

adihrustic commented:
> Cant you still close through the notification?
Yes. But that movement is more awkward using one hand than moving the player to the close button.
